### PR TITLE
[KYUUBI #3052] Support to get the real client ip address for http connection when using VIP as kyuubi server load balancer

### LINKED ISSUE #3052: [FEATURE] Support to get real ip for http connection when using VIP as kyuubi server LB #3051
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the feature

We deploy kyuubi server on k8s and using k8s lb and the connection path is:

client -> lb -> kyuubi server.

Now the client ip address is always the lb ip and is meaning less.

### Motivation

_No response_

### Describe the solution

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

To close #3052 

If your server is behind a load balancer or other proxy, the server will see this  load balancer or proxy IP address as the client IP address, to get around this common issue, most load balancers or proxies offer the ability to record the real remote IP address in an HTTP herader that will be added to the request for other devices to use.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Review by pan3793 — APPROVED** [review]
The change LGTM, please attach a manual test result in the PR description.

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 1]
```diff
@@ -580,6 +580,17 @@ object KyuubiConf {
       .booleanConf
       .createWithDefault(true)
 
+  val FRONTEND_PROXY_HTTP_CLIENT_IP_HEADER: ConfigEntry[String] =
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:583]: why we need this conf?

**Comment by yaooqinn** [comment]
this is odd, shall it be `forwardedAddresses/X_FORWARDED_FOR_HEADER`

**Review by yaooqinn — CHANGES_REQUESTED** [review]
this is a risk and nonstandard change for me

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala (hunk 1 above)
Comment (turboFei) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:583]: It is a config for the proxy.

I am using ebay hlb, the header is `X-eBay-Web-Tier-IP`.

**Comment by turboFei** [comment]
I tried to 

> this is odd, shall it be `forwardedAddresses/X_FORWARDED_FOR_HEADER`
> 
> > X-Forwarded-For
> > The X-Forwarded-For (XFF) request header is a de-facto standard header for identifying the originating IP address of a client connecting to a web server through a proxy server.

In fact, it depends on the proxy setting.

For the ebay HLB, there is no ` X-Forwarded-For` header.

And the header to present the real client ip is `X-eBay-Web-Tier-IP` instead of `X-Real-IP`.

**Comment by turboFei** [comment]
FYI:

https://docs.fastly.com/signalsciences/faq/real-client-ip-addresses/#setting-alternative-headers-directly-in-the-agent


Because the LB I am using does not have `X-Forwarded-For`, so I can not make manual testing for that.

How about do it in another pr.

**Comment by yaooqinn** [comment]
how about making `X-Forwarded-For/X-Real-IP` in one and delivering with forwardedAddresses

we need the IP of the direct accessor if it is not only proxying address but also proxying user

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
on second thought, it makes sense. please add a ut

**Comment by turboFei** [comment]
ok, I will add UT for rest connection.

**Comment by turboFei** [comment]
thanks @yaooqinn @pan3793 
merged to master

**Pull request closed without merging**